FileSystem/FileMapperMocker.js

```diff
@@ -10,28 +25,490 @@ var USE_FILEMAPPER_MOCKER = true;
+  getFileId: function(currentDirectory, path, type) {
     if (path in this.mappings) {
       return this.mappings[path];
     }
     throw new Error(path + ' not mapped!');
   },
-  registerFile: function (currentDirectory, path, fileId) {
+  registerFile: function(currentDirectory, path, fileId) {
     this.mappings[path] = fileId;
   },
-  hasMapping: function (currentDirectory, path, type) {
+  hasMapping: function(currentDirectory, path, type) {
     return path in this.mappings;
   },
 };
 
-var FileMapper = USE_FILEMAPPER_MOCKER ? VBAFileMapperMocker : VBAFileMapper;
-
 function createFile(currentDirectory, path, mimeType) {
   var fileId = DriveApp.createFile(path, '', mimeType).getId();
   FileMapper.registerFile(FileSystem.currentDirectory, path, fileId);
   return fileId;
 }
+
+/**
+ * Mocks the FileMapper module. The mocker is implemented as we need a way to
+ * test the FileSystem APIs before the FileMapper is ready. This is to be
+ * replaced with complete implementation. This mocker implements all the
+ * functionality as per the design of the FileMapper. The primary difference is
+ * that there is no config file, instead there is a hardcoded base directory.
+ * All file paths have to contain this base directory as prefix. The mocker does
+ * NOT do the following things 1) Check MimeType, Permissions 2) Check for
+ * duplicate files 3) Throw errors as per design (because we are not doing
+ * checks) 4) Prompt user in case file is not found
+ */
+var VBAFileMapperMocker2 = {
+  baseDirectory: 'c:\\user\\desktop',
+  getFileId: getFileId,
+  getFolderId: getFolderId,
+  hasFile: hasFile,
+  hasFolder: hasFolder,
+  deleteFile: deleteFile2,
+  deleteFolder: deleteFolder,
+  createFile: createFile2,
+  createFolder: createFolder,
+  moveFile: moveFile,
+  moveFolder: moveFolder,
+  copyFile: copyFile,
+  copyFolder: copyFolder,
+  findFilesByPattern: findFilesByPattern,
+  findFoldersByPattern: findFoldersByPattern,
+  getRelativePathSplit: getRelativePathSplit,
+  getBaseDriveFolder: getBaseDriveFolder,
+};
+
+var FileMapper = USE_FILEMAPPER_MOCKER ? VBAFileMapperMocker2 : VBAFileMapper;
+
+/**
+ * Get File API
+ * Gets the file id of a local file
+ * @param {string} localPath Local filepath of the file
+ * @return {string} File id of the local file
+ */
+function getFileId(localPath) {
+  // Get platform agnostic relative path split
+  var pathSplit = this.getRelativePathSplit(localPath);
+  // Get base directory
+  var root = this.getBaseDriveFolder();
+  // Find folder containing local file
+  for (var i = 0; i < pathSplit.length - 1; i++) {
+    var folderIterator = root.getFoldersByName(pathSplit[i]);
+    if (!folderIterator.hasNext()) {
+      throw new Error('File not found');
+    }
+    root = folderIterator.next();
+  }
+  // Obtain the file
+  var fileIterator = root.getFilesByName(pathSplit[pathSplit.length - 1]);
+  if (!fileIterator.hasNext()) {
+    throw new Error('File not found');
+  }
+  // Return file id
```

Comment:
Can remove this comment as it is obvious from the code

Reply:
Fixed